Source/WebCore/rendering/GlyphDisplayListCache.h

```diff
@@ -41,8 +43,53 @@ namespace InlineDisplay {
 struct Box;
 }
 
+class GlyphDisplayListCacheBase {
+public:
+    unsigned size() const
+    {
+        return m_entriesForTextRun.size();
+    }
+
+    size_t sizeInBytes() const
+    {
+        size_t sizeInBytes = 0;
+        for (auto& entry : m_entriesForTextRun)
+            sizeInBytes += entry.value->displayList().sizeInBytes();
+        return sizeInBytes;
+    }
+
+protected:
+    class Entry : public RefCounted<Entry>, public CanMakeWeakPtr<Entry> {
+        WTF_MAKE_FAST_ALLOCATED;
+    public:
+        static Ref<Entry> create(std::unique_ptr<DisplayList::InMemoryDisplayList>&& displayList, std::optional<FloatSize> relevantScaleFactor)
+        {
+            return adoptRef(*new Entry(WTFMove(displayList), relevantScaleFactor));
+        }
+
+        DisplayList::InMemoryDisplayList& displayList() { return *m_displayList.get(); }
+        const std::optional<FloatSize>& relevantScaleFactor() const { return m_relevantScaleFactor; }
+
+    private:
+        Entry(std::unique_ptr<DisplayList::InMemoryDisplayList>&& displayList, std::optional<const FloatSize> relevantScaleFactor)
+            : m_displayList(WTFMove(displayList))
+            , m_relevantScaleFactor(relevantScaleFactor)
+        {
+            ASSERT(m_displayList.get());
+        }
+
+        std::unique_ptr<DisplayList::InMemoryDisplayList> m_displayList;
+        std::optional<FloatSize> m_relevantScaleFactor;
+    };
+
+    enum class Sharing { No, IfScaleFactorMatches, Yes };
```

Comment:
I wonder how common it will be to use the cache when the scale factor does not match? Maybe we don't need to handle this case? But this also makes me think that the cache should take ShouldSubpixelQuantizeFonts into account.

Reply:
I don't think it will be common. I'll make the cache always check for a scale factor match, instead of doing that only if there are no images among the glyphs. (And I'll add the subpixel quantization state to that check too.)